src/web/routes/application/register-form-routes.js

```diff
@@ -18,13 +18,28 @@ const handleOptionalMiddleware = (operation, fallback = middlewareNoop) => {
   return operation
 }
 
-const createRoute = (csrfProtection, steps, router) => (step) =>
+const getPreviousPage = (steps, step) => {
+  const index = steps.indexOf(step)
+
+  if (index === -1) {
+    throw new Error(`Unable to find ${step} in the list of steps`)
+  }
+
+  // first page doesn't have a back link
+  if (index > 0) {
+    const previousStep = steps[index - 1]
+    return previousStep.path
+  }
+}
+
+const createRoute = (csrfProtection, steps, router) => (step) => {
+  const previous = getPreviousPage(steps, step)
   router
     .route(step.path)
     .get(
       csrfProtection,
       getSessionDetails,
-      renderView(step.template, step.pageContent, step.next)
+      renderView(step.template, step.pageContent, step.next, previous)
```

Comment:
I feel like adding `previous` is making the api / surface area for `renderView` too big. I think it would be better to create another middleware step (maybe `configureGet`??) and add the back link to `res.locals`, instead of passing it to the template explicitly in `res.render()`.